### ISSUE TITLE
Gradient compute fail when using RawFeatureVector

### ISSUE THREAD

**Issue by saalimon** [Qiskit/qiskit-machine-learning#95]
### Information

- **Qiskit Machine Learning version**: 0.1.0
- **Python version**: 3.7.10
- **Operating system**: All

### What is the current behavior?
This message occurs when using `RawFeatureVector` with `CircuitStateFn`
`Cannot compute gradient operator! Continuing without gradients!`

### Steps to reproduce the problem

```python
from qiskit import QuantumCircuit
from qiskit.opflow import StateFn, PauliSumOp, AerPauliExpectation, Gradient
from qiskit.utils import QuantumInstance
from qiskit_machine_learning.neural_networks import OpflowQNN
from qiskit_machine_learning.circuit.library import RawFeatureVector

expval = AerPauliExpectation()
gradient = Gradient()
qi_sv = QuantumInstance(Aer.get_backend('statevector_simulator'))

inputs = RawFeatureVector(16)
qc = QuantumCircuit(4)
qc.append(inputs,range(4))
qc_sfn = StateFn(qc)
H1 = StateFn(PauliSumOp.from_list([('Z', 1.0)]))
op = ~H1 @ qc_sfn
qnn4 = OpflowQNN(operator = op, input_params = inputs.parameter, weight_params = [], exp_val = expval, gradient = gradient, quantum_instance = qi_sv)

```

### What is the expected behavior?
Can pass the `Parameter` or `ParameterVector` to `OpflowQNN` or another Opflow from `qiskit_machine_learning.neural_networks`

### Suggested solutions
Check the Gradient condition on `qiskit.opflow`

**Comment by ElePT** [comment]
Hello! I would like to work on this issue.

**Comment by adekusar-drl** [comment]
Closed in #182.
